File: FE/WEB/app/models/template.js
import Model, { attr, belongsTo } from '@ember-data/model';

export default class TemplateModel extends Model {
  @belongsTo('category', { async: true, inverse: 'template' }) category;

  @attr('string', { defaultValue: '' }) name;
  @attr('string', { defaultValue: '' }) description;
  @attr('string', { defaultValue: '' }) objectText;
  @attr('string', { defaultValue: '' }) content;
  @attr('string', { defaultValue: '' }) contentNoHtml;
  @attr('number', { defaultValue: 0 }) categoryId;
  @attr('boolean', { defaultValue: false }) active;
  @attr('string', { defaultValue: '' }) tags;
  @attr('string', { defaultValue: '' }) code;
  @attr('string', { defaultValue: '' }) language;
  @attr('boolean', { defaultValue: false }) erasable;
  @attr('boolean', { defaultValue: true }) copyInNewTenants; // indica se va copiato nei nuovi tenant

  @attr('boolean', { defaultValue: false }) erased;
  @attr('number', { defaultValue: 0 }) order;
  @attr('string', { defaultValue: '' }) featuredImage;
  @attr('arrays', { defaultValue: '{}' }) freeField;
  @attr('date', {
    defaultValue() {
      return new Date();
    },
  })
  createdAt;
}

File: BE/Src/MIT.Fwk.Infrastructure/Entities/Template.cs
using JsonApiDotNetCore.Resources;
using JsonApiDotNetCore.Resources.Annotations;
using System;
using System.ComponentModel.DataAnnotations.Schema;

namespace MIT.Fwk.Infrastructure.Entities
{
    [Table("Template")]
    public class Template : Identifiable<string>
    {


        [Attr] public string Name { get; set; }
        [Attr] public string Description { get; set; }
        [Attr] public string Content { get; set; }
        [Attr] public string ContentNoHtml { get; set; }
        [Attr] public int CategoryId { get; set; }
        [Attr] public bool Active { get; set; }
        [Attr] public string Tags { get; set; }
        [Attr] public string Code { get; set; }
        [Attr] public string Language { get; set; }
        [Attr] public string ObjectText { get; set; }
        [Attr] public string FeaturedImage { get; set; }
        [Attr] public bool Erasable { get; set; }
        [Attr] public bool CopyInNewTenants { get; set; }
        [Attr] public bool Erased { get; set; }
        [Attr] public int Order { get; set; }
        [Attr] public string FreeField { get; set; }
        [Attr] public DateTime? CreatedAt { get; set; }

        [HasOne]
        public virtual Category Category { get; set; }
    }
}
